core/lib/spree/event/execution.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Spree
+  module Event
+    # Execution of a {Spree::Event::Listener}
+    #
+    # When an event is fired, it executes all subscribed listeners. Every single
+    # execution is represented as an instance of this class. It contains the
+    # result value of the listener, along with helpful metadata as the time of
+    # the execution or a benchmark for it.
+    #
+    # You'll most likely interact with this class for debugging or logging
+    # purposes through the returned value in {Spree::Event.fire}.
+    class Execution
+      # The listener to which the execution belongs
+      #
+      # @return [Spree::Event::Listener]
+      attr_reader :listener
+
+      # The value returned by the {#listener}'s block
+      #
+      # @return [Any]
+      attr_reader :result
+
+      # Benchmark for the {#listener}'s block
+      #
+      # @return [Benchmark::Tms]
+      attr_reader :benchmark
+
+      # Time of execution in UTC standard
+      #
+      # @return [Time]
+      attr_reader :utc_time
+
+      # @private
+      def initialize(listener:, result:, benchmark:)
+        @listener = listener
+        @result = result
+        @benchmark = benchmark
+        @utc_time = Time.now.utc
```

Comment:
Also, instead of `@utc_time`, can we call this something more meaningful, e.g. `executed_at` or similar?

Reply:
I renamed it to `execution_time`, and I also renamed `Spree::Event::Event#utc_time` to `#firing_time`.

I agree that the name is more meaningful. The decision for the old name came from this conversation on a previous closed PR:

https://github.com/nebulab/solidus/pull/148#discussion_r671023680

However, now that the time can be injected, it no longer needs to be a `UTC` value. And, anyway, the information is within the value, so I don't think it's that important.